### ISSUE TITLE
Run args not correctly logged in hyperparameters view

### ISSUE THREAD

**Issue by ivan94fi** [comet-ml/issue-tracking#452]
## Describe the Bug
The command line arguments passed to the python script are sometimes not correctly logged as `run_arg_<number>`, but are just logged as hyperparameters. In my tests, adding a long option causes this bug, but this might be coincidental.

The experiment constructor is called like this:
```python
experiment = Experiment(
    project_name="...",
    auto_metric_logging=False,
    log_graph=False,
    disabled=config.no_comet,
    auto_output_logging="simple",
    log_git_patch=False,
    display_summary_level=0,
)
```
and the parameters are logged like so:
```python
config: argparse.Namespace = get_config()  # parse the CLI arguments
experiment.log_parameters(vars(config))
```
The cli arguments of interest are: `-w 12 train -e 80 --pixel-shuffle`, and are translated in the following hyperparameters, without any "run arg" (please ignore any other hyperparameter, they all have default values which are not changed in these runs):

**Screenshot 1:**
![Screenshot from 2021-12-13 14-47-36](https://user-images.githubusercontent.com/30447023/145824765-d7c3d204-57e0-4f39-905d-2e40863597f0.png)

**Screenshot 2:**
![Screenshot from 2021-12-13 14-47-49](https://user-images.githubusercontent.com/30447023/145824799-713ee1ec-b2cb-4a4e-b729-184ab130d2f8.png)

**Screenshot 3:**
![Screenshot from 2021-12-13 15-50-07](https://user-images.githubusercontent.com/30447023/145834268-9e454a69-d599-4045-82f2-50c607951fae.png)

As can be seen in these images, all the mentioned options, `workers`, `pixel-shuffle` and `epochs` are repeated (`e` is the short version of `epochs`, `w` is the same for `workers`), as both the cli option and the actual hyperparameter in the parsed argparse.Namespace are logged . Actually the `pixel-shuffle` option ends up picking a wrong value, which should be assigned to the `command` option (not showed in screenshots).

## Expected behavior
All the run args should be logged, together with the corresponding parameter: the arguments `-w 12 train -e 80`, that is, the same as before but without the `pixel-shuffle` long option, are correctly logged, and run args appear as expected:
![Screenshot from 2021-12-13 15-57-56](https://user-images.githubusercontent.com/30447023/145835565-114e24fe-a5df-4065-83d1-e1de85aa4662.png)

From these tests it seems that adding the `--pixel-shuffle` option actually causes the bug, but, again, this does not appear as a sound conclusion, just an observation.

## Where is the issue?
- [x] Comet Python SDK
- [x] Comet UI
- [ ] Third Party Integrations (Huggingface, TensorboardX, Pytorch Lighting etc)

This issue seems to appear with the last version of the comet web UI, but the older experiments still show the run args correctly in the new UI, so this may be a coincidence and maybe it is a bug in the python SDK.

The issues appear both with python SDK version 3.14.1 and the latest version 3.23.0 (I tested only these two versions). 

## To Reproduce
[EDIT] With this script I am able to reproduce the bug independently from my project:
* `python test.py -a a train -e e`: correct hyperparameters logged.
* `python test.py -a a --bool-argument train -e e`: wrong hyperparameters logged

So after all maybe it is in fact the boolean argument (declared with `action="store_true"`) that causes the problem.

`test.py`:
```python
from comet_ml import Experiment
from argparse import ArgumentParser

parser = ArgumentParser()

parser.add_argument("-a", "--argument-a")
parser.add_argument("-b", "--argument-b")
parser.add_argument("-c", "--argument-c")
parser.add_argument("--bool-argument", action="store_true")

subparsers = parser.add_subparsers(
    title="Subparsers",
    description="Description",
    dest="command",
    required=True,
)
train_parser = subparsers.add_parser("train")
train_parser.add_argument("-d", "--argument-d")
train_parser.add_argument("-e", "--argument-e")
train_parser.add_argument("-f", "--argument-f")

config = parser.parse_args()

experiment = Experiment(
    project_name="test",
    auto_metric_logging=False,
    log_graph=False,
    auto_output_logging="simple",
    log_git_patch=False,
    display_summary_level=0,
)

experiment.log_parameters(vars(config))
```

**Comment by DN6** [comment]
Thanks for raising this @ivan94fi and providing so much context. Let me take a look at what's going on here. I will get back to you as soon as possible.

**Comment by ivan94fi** [comment]
Thank you @DN6 for the quick reply. I updated the bug report with a script to reproduce the bug. Hope it helps.

**Comment by DN6** [comment]
Hello @ivan94fi . I was able to reproduce your issue. I believe boolean flags and subparsers can cause issues with our autologger's command line parser.

Another thing I noticed is that you are logging the parsed args with `experiment.log_parameter` but you have not disabled the autologger's command line parser. Can you update your Experiment object with the following configuration and try rerunning your script to see if it produces the desired output

```
experiment = Experiment(
    project_name="test",
    auto_metric_logging=False,
    log_graph=False,
    auto_output_logging="simple",
    log_git_patch=False,
    display_summary_level=0,
    parse_args=False,
)
```

**Comment by DN6** [comment]
Hi @ivan94fi. Just so I understand your use case correctly. You would like to log the parsed args, and also log the run args as separate parameters in order to track which args were modified at runtime?

**Comment by ivan94fi** [comment]
Hi @DN6.

> You would like to log the parsed args, and also log the run args as separate parameters in order to track which args were modified at runtime?

> The parsed args logged as parameters should reflect the values passed in at runtime?

Yes, these are correct. As a matter of fact I just want the arguments string (`sys.argv`) to be logged somewhere, and as Comet itself logged that in the Hyperparameters section, I just sticked with this convention, which seems reasonable.

The rationale behind this use case is what you mentioned: I want to be able to see all the hyperparameters, and also which parameter is changed at runtime when executing the experiment.
